### DISCUSSION TITLE
MAF bench design plans

### DISCUSSION THREAD

**Discussion by DeeEmm** [DIY-Flow-Bench/DIY-Flow-Bench#99] [Flow Bench Design]
<img width="1645" alt="Screen Shot 2021-09-22 at 6 00 54 pm" src="https://user-images.githubusercontent.com/3038710/134311851-f4882079-6473-4f63-aec1-3a762708afdc.png">

I've started to put together some plans / drawings for a basic MAF style shop-vac powered flow bench. This is the flow bench in its most basic form and is a starting point for anyone wanting to build their own bench using the DIYFB code / shield.

It's not going to be the ultimate in flow bench performance as the aim is simply to provide an easy to source and assemble design. So of course improvements are easy to make, but it makes a good base from which to start.

The design is based on experiments that we conducted as well as drawing on the knowledge of others who have trodden this path before. I did document this basic design in the docs section but thought is prudent to sketch the design out so that it was easier to understand. These sketches will form part of a more formal design that will be provided in the docs folder.

I'll add to this discussion as I progress

**Comment by DeeEmm** [comment]
WIKI updated - https://github.com/DeeEmm/DIY-Flow-Bench/wiki/Hardware-Design.

**Comment by Captain-Beaky** [comment]
OK, I'm finally getting my act together and putting together a flow bench. For the moment, I'm sticking with the V1 software, as it works, and I don't want to get into the complication of another computer hanging around and maybe getting damaged while I'm doing the physical build of the bench. (and I can't readily get my hands on an ESP32 out here that quickly anyway)...

Sorry if this is a bit long, but I want to understand the essentials of the flow bench, and work out what is most important.
I would rather only build this once, and get it as right as possible, so I'll list what i think is important, ask questions, and hope that someone corrects me or adds more info, so I can come up with a set of constraints and important info. This is for the actual air-flowing components - not the electronics or anything else just now....

So, in a flow bench, in the air-flow path, we have:-
1 The test port pipe
2 The test port adapter
3 Pipework
4 MAF
5 Pipework
6 The Vacuum / Pressure source.

I'll go through what I think is important and ask questions...

1. Test port inner diameter. We will be placing a cylinder head face down on this test port, and the test port is essentially mimicking the cylinder bore, so we want a diameter that will be larger than the cylinder size we will be using. I work on a lot of big Jags and Land Rovers, with bore sizes around 96 - 100mm, so the commonly available 125mm OD (114mm ID) drain pipe would give me plenty of room. (Not many engines are more than 110mm bore size, and many multi cylinder bikes are tiny by comparison).
2. Test Port Adapter - we need this to convert the Test port size down to whatever we are testing - would this be a flat plate the size of the cylinder bore ID, or would it be more like a top hat sleeve that properly imitates the bore size - effectively reducing the test port bore? How important is this? (I'm guessing that it needs to be a sleeve? but how long?)
3. We have some pipework that joins the test port to our chosen MAF - I'm guessing this needs to either be a minimum length or have some features in it to ensure laminar flow into the MAF ? Common on some airflow test gear I have seen is a length of pipe with what looks suspiciously like a length of connecting pipe with drinking straws filling it, to eliminate turbulence. What do people use?
4. MAF - we have many available, I'm choosing the biggest (largest flow) that I can get the data for.
5. Pipework from MAF to vac/pressure source - again, this will need some element in it to reduce turbulence?
6. Pressure vac source - this has to flow the right amount of air at the right pressures - most likely to be a multiple of vacuum cleaner motors. 
6a, But, the whole system has to be reversible - it has to suck or blow, so the pipework has to be switchable (and maybe the MAF orientation changed too)

One current train of thought is to have the MAF and vacuum source set up in line so the MAF is always on the "vac" side, and we don't have to mess with it's direction when we convert from intake to exhaust testing.

So for the intake testing, the air path would be:-
Test port - pipe - MAF - Pipe - VAC (Blower) Pressure - atmosphere

But for exhaust testing, the MAF and blower assy would be reversed. it would be:-
Test Port - Pipe - Pressure (Blower) Vac - MAF - Pipe - atmosphere

The obvious catch is that the MAF would be before the blower, so absolute care must be taken to ensure there are no leaks.

Any other issues anyone can think of?

One possible advantage of this is that the pipework in front of the MAF only needs to be optimised to ensure laminar flow into one side of the MAF, so it could be bigger / more effective.

Also, MAF's are really designed to be before any pressure generating component of the intake - they run from atmospheric down to a small vacuum - usually just after the air filter. I'm not so sure if there is any downside to running one on the pressure side of the blower? 

But - the blowers will create heat, and reduce the density of the air. So I'm guessing we'd need to correct for this? (on the vac side it would be ambient temp, on the pressure side, it would be the elevated temp...

I need to perhaps think about that a bit more - it's 23:30 here, and I've had a few drinks, so I'll leave this right here for now.

Cheers (and Merry Christmas)

Mike.

**Comment by DeeEmm** [comment]
Hi Mike.

We did some experiments with laminar flow correctors (drinking straws) and found that it did not really make a difference to the flow figures.

Theoretically after any change in diameter or any bend in the pipe the gasses require a distance to settle back into a laminar flow pattern. This distance is calculable and for this particular application ends up at something like 4 metres which is simply impractical to use due to space constraints.

The same issue also affects cars, where in most engines it's hard to find 4cm space to have a straight section let alone 4 metres. So to overcome this most MAFs actually have flow correctors built into them. The flow correctors are generally a fine honeycomb filter mesh but in some cases are just a basic plastic mesh. (check any MAF you intend to use)

The fine honeycomb style are far superior, in fact if you turn the filter slightly away from you as you look through it you will note that you will not be able to see through it once the filter is a few degrees from perpendicular. These are very good at straightening the flow enough to get an accurate reading in the constraints of an engine bay, which is all that is required and quite probably the reason that the drinking straw experiment did not net the results we expected (as the air flow was already corrected)

The other big consideration is a 90 degree bend between your vac pump and the main straight to convert any vortexes into laminar flow. This is especially true for large single impeller air pumps which can create a spinning vortex. 

If you are building a more traditional vac source with multiple vac motors, then you might want to consider a stand alone vac box which  has a port at each end and can be easily rotated to change from vac to blow. This eliminates the need for complex diverters and flaps. but of course is a little more involved to swap over than turning a simple lever as per a more traditional bench. Bench design really comes down to personal choice. Everyone has their own preferences.

Regarding reversing MAF sensors, you can also use cartridge style sensors that mount directly into the pipe. These will work with any size of pipe and have the added advantage that they can be easily rotated when you want to reverse the flow direction. The caveat with this is that it is harder to get published flow data for the diameter you need. Plus you either need to figure out a way to add flow correctors both up and downstream of the sensor or run without them.

There's still some experimentation to do with the cartridge style sensors to validate them but in theory they should work as described. Else it's just a case of using a regular MAF and changing its orientation in the pipe when you want to run reverse flow.  

Regarding the flow data for a cartridge style sensor being used in a different pipe diameter to that in the published data, you can convert it using a spreadsheet but the bench itself can be used to map the sensor characteristics for any sensor provided that a known sensor is used as a reference, but this ideally needs to be done under controlled conditions to ensure data accuracy. This is essentially how most MAFs are set up and mapped. 

Leaks are your enemy. Which is why it is important to perform a leak test each time any measurements are taken. Essentially every time you set a job up you need to do a leak test.

Most benches use an adaptor that sits on top of the bench. This is is the diameter of the cylinder that is being tested. Often they are clear so that you can see the flow entering the cylinder using flow tools - take a look at a professional bench and you will see similar setups. You'll need to make or purchase a similar device.

Air temp is not an issue due to the way that the MAF works

You can also run the MAF with pressure from the vac source with no issue as the pressure is negligible (air density is also factored into the calcs)

I would not run the MAF and blower as you have suggested. I would advise to keep the same configuration but simply reverse the flow and flip the MAF. This is not to say that what you have suggested would not work but would require further investigation.

Hope this helps.

**Comment by Captain-Beaky** [comment]
Thanks for that - I’m off to buy some lengths of drain pipe!

**Reply by Captain-Beaky** [reply]
Yes, I did the calculations, and the “optimum length” of pipes reads more like the dimensions of the room I’ve set aside to put this equipment in!

There’s a good video from Bowling and Grippo where the test bench pipework runs all the way around the room to get nice laminar flow. 

Also, in an engine bay install, laminar flow is less important, as long as the flow meter runs in a stable configuration. The actual perfect mass flow measurement isn’t so important, as it’s correctable in the mapping.

Yes, multiple vacuum cleaner motors in a blower box - cheap and simple. Also, I found some that have separate cooling airflows - should help temperature control of the motors, and reduce one thing we have to correct for.

Yes, of course, I remember seeing the clear test adapters. Silly me. 
I’ll be making some up myself to start with. 
Unless I stumble on a reasonably priced source for large bore clear tube.

**Reply by robssmv2** [reply]
I am effectively converting an orifice tube bench to have an integrated MAF. My idea was to make a flow offset calculation table using a series of calibration plates. 

I don't think it will matter that much, but, its something to maybe check once and see whether or not its even required.. 

Also,  I was going to design a cartridge style MAF tube whereby the maf could be rotated in the tube assembly, with an O ring and a snap ring and the entire assembly inside the flowbench. 

I don't know if everyone is thinking an orifice plate bench is better or? I even debated about perhaps just utilizing the MAF as a sort of backup for the orifice reading? Some ideas I was playing with in branching the code and implementing some extra features..

**Reply by DeeEmm** [reply]
@robssmv2 There is already provision for a cartridge based MAF and there are also some 3D print designs for the mount in the 3D folder....

https://github.com/DeeEmm/DIY-Flow-Bench/blob/master/3D%20Parts/110mm%20Dia%20MAF%20Mount%20v4.stl

The code also allows you to scale up the pipe diameter to increase the flow range of the MAF. The only caveat is that you need to know the baseline MAF diameter for the scaling to work. If you can get hold of a cartridge version of any of the MAFs that we have validated data for then it can be used to scale up to a larger diameter.

> Also, in an engine bay install, laminar flow is less important, as long as the flow meter runs in a stable configuration.  The actual perfect mass flow measurement isn’t so important, as it’s correctable in the mapping.
>

This is incorrect. The MAP is a lookup table that is driven by the MAF + RPM inputs. It does not provide compensation for the MAF 

The MAF is the sole input into the system that determines engine load and therefore what fuelling is required. It is actually very accurate and has a very fast response. 

The simplistic way to look at the MAF is that it can very accurately control injectors that fire at rates of between 15 to over 70 times a second, controlling the precise amount of time that the injector needs to be opened, delivering a perfectly metered amount of fuel. It can do this very accurately over the entire RPM range of the engine. 

It's use in measuring air flow in this application is far superior to differential measuring. This becomes even more obvious when you consider that no modern cars use differential pressure measurement to meter fuelling as part of engine management. Well this is not strictly true. There was a time when all cars used to use a device called a carburettor, which uses differential pressure to meter the fuel based on airflow. But we all know how that ended. lol

Is one system better than the other?

MAF sensors measure flow on the molecular level and are inherently accurate, but conditions need to be controlled. 

Both MAF benches and DP benches suffer from the same issue. Shit in = Shit out. I believe that a MAF based system can exceed current commercial DP benches in accuracy and repeatability as there are many inherent flaws in the 'ratiometric' principle of measurement currently being used, and the way that is is implemented. But unfortunately the only way to prove this, is to do it.

Flaws in DP benches were highlighted in the PAP tests that were conducted some 15 years back where a set of orifice plates were passed around between flow bench owners and the result compared

![PassAroundPlatesResults](https://github.com/user-attachments/assets/ce326943-89ef-40bf-be1e-1b276b73ff17)

The results show staggering differences, basically placing a big question over vendors claims of accuracy.

Would love to repeat the same test with a bunch of home made MAF benches to see how the results stack up. :D

**Comment by DeeEmm** [comment]
I forgot to mention something...

There is also another way to reverse the flow which is a bit of a hack but if you are just after validating flow improvements is absolutely fine. Simply mount the exhaust port flange to the bench inlet (instead of the cylinder) so that you are sucking the air through the port in the correct direction.

The dynamics are obviously different from the actual application (suck instead of blow) but when you consider that actual exhaust flow leaving the cylinder is supersonic and of a vastly different temperature profile the differences introduced by sucking through the port instead of blowing could almost be considered negligible.

Things like these need validation of course. This is not a new suggestion, people have been doing this for years, but validating measured flow against a proper bench is an experiment worth carrying out and quantifying.

Regarding inlet temp. It is only a consideration for converting from mass flow to volumetric flow and then only requires the ambient temp. The bench is calibrated to a known source so most small anomalies are already accounted for

However, it is an interesting consideration...

Theoretically we should be able to compare the calculated flow figure for the calibration orifice for vac and pressure (reverse flow). These figures should be the same. So any introduced anomaly or deviation created by increased air temperatures from the vac source, can be cancelled out.

I will add a consideration for reverse flow calibration to the road map. Then we can undertake some experiment and validate whether it is required. The hardest part of that particular test would be getting a good enough seal for the test orifice as it would have a tendency to want to blow it odd of the top of the test platten.

If you do make / find some clear tube please share. That's one part I have not figured out yet.

/DM
